modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/registry/AppFacRemoteRegistryAccessService.java

```diff
@@ -531,8 +545,138 @@ private AppFactoryResource convertResourceObject( AppFacResource appFacResource)
+     * @param applicationId id of the application
+     * @param stage         stage of the application
+     * @param key           property key
+     * @param value         property value
+     * @param isSecured     property is secured or not
+     * @param version       version of the application
+     */
+    @Override
+    public void addRuntimeProperties(String applicationKey, String stage, String key, String value,
+            String isSecured, String version) {
+
+        RuntimeProperty runtimeProperty = new RuntimeProperty();
+        Map<String, String> properties = new HashMap<String, String>();
+
+        boolean isChecked = Boolean.parseBoolean(isSecured);
+
+        //Checked whether the property type is secured
+        if (isChecked) {
+            runtimeProperty.setPropertyType(RuntimeProperty.PropertyType.SENSITIVE);
+            runtimeProperty.setName(key);
+            byte[] encodedValue = Base64.encodeBase64(value.getBytes(Charset.forName("UTF-8")));
+
+            properties.put(key, new String(encodedValue, Charset.forName("UTF-8")));
+            runtimeProperty.setProperties(properties);
+
+        } else {
+            runtimeProperty.setPropertyType(RuntimeProperty.PropertyType.ENVIRONMENT);
+            runtimeProperty.setName(key);
+
+            properties.put(key, value);
+
+            runtimeProperty.setProperties(properties);
+
+        }
+
+        List<RuntimeProperty> runtimeProperties = new ArrayList<RuntimeProperty>();
+        runtimeProperties.add(runtimeProperty);
+
+        ApplicationContext applicationContext = new ApplicationContext();
+
+        applicationContext.setCurrentStage(stage);
+        applicationContext.setId(applicationKey);
+        applicationContext.setVersion(version);
+
+        TenantInfo tenantInfo = new TenantInfo();
+
+        int tenantId = PrivilegedCarbonContext.getCurrentContext().getTenantId();
+        String tenantDomain = PrivilegedCarbonContext.getCurrentContext().getTenantDomain();
+
+        tenantInfo.setTenantId(tenantId);
+        tenantInfo.setTenantDomain(tenantDomain);
+
+        applicationContext.setTenantInfo(tenantInfo);
+
+        KubernetesRuntimeProvisioningService kubernetesRuntimeProvisioningService
+                = new KubernetesRuntimeProvisioningService(applicationContext);
+
+        try {
+            DeploymentConfig deploymentConfig = JDBCResourceDAO.getInstance().getDeploymentConfig(applicationKey, stage);
+            kubernetesRuntimeProvisioningService.setRuntimeProperties(runtimeProperties, deploymentConfig);
+        } catch (RuntimeProvisioningException e) {
+            String message = "Unable to create runtime property : " + key + " for application "
+                    + applicationKey + " in tenant " + tenantDomain;
+            log.error(message, e);
+        } catch (AppFactoryException e){
+            String message = "Unable to getting deployement config details from the database";
+            log.error(message, e);
```

Comment:
Why you are not throwing any exceptions here? How does the calling method/client know whenever something is wrong?